extensions/vscode/src/multiStepInputs/newDeployment.ts

```diff
@@ -490,21 +756,28 @@ export async function newDeployment(
     };
 
     // start the progression through the steps
-    await MultiStepInput.run((input) => inputEntryPointSelection(input, state));
+    await MultiStepInput.run((input) =>
+      inputEntryPointFileSelection(input, state),
+    );
     return state as MultiStepState;
   }
 
   // ***************************************************************
   // Step #1 - maybe?:
   // Select the entrypoint to be used w/ the contentRecord
   // ***************************************************************
-  async function inputEntryPointSelection(
+  async function inputEntryPointFileSelection(
     input: MultiStepInput,
     state: MultiStepState,
   ) {
+    // in case we have backed up from the subsequent check, we need to reset
+    // the array that it will update. This will allow steps to be the minimum number
+    // as long as we don't know it will take another one.
+    inspectionResults = [];
+
     // skip if we only have one choice.
     if (entryPointListItems.length > 1) {
```

Comment:
This no longer works how we anticipate. I noticed while working in the `fastapi-simple` directory with only the `simple.py` file as an entrypoint I was still getting the file selection prompt.

I debugged it a bit, and it looks like the "Open Files" label is part of the array.

Perhaps we need to keep a separate array of entrypoints rather than looking at the items or something else?

![CleanShot 2024-07-31 at 16 28 21](https://github.com/user-attachments/assets/af43f1ca-938e-454d-8198-0d8097baabc3)

Reply:
This has been resolved. See commit: https://github.com/posit-dev/publisher/pull/2051/commits/40b9a0ad82d8887725ca07cc9e701bc3a1946179